Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -214,6 +214,25 @@ open class ChatChannelListVC: _ViewController,
         chatChannelListLoadingView.updateContent()
     }
 
+    /// Replaces the channel list query and loads the new data.
+    ///
+    /// - Parameter query: The new channel list query.
+    public func replaceQuery(_ query: ChannelListQuery) {
+        let newController = controller.client.channelListController(
+            query: query
+        )
+        replaceChannelListController(newController)
```

Comment:
Shouldn't we maintain the filter closure of the existing controller, here?

Otherwise we may want to mention that this method will result in the creation of a new controller with a nil filter closure.

Reply:
This is why the `replaceChannelListController` exists 👍 We can't reuse the filter block for now because it is a private property. 

The `replaceQuery` will be sufficient for most cases, but if a customer wants more control, they can use `replaceChannelListController`.

There a 2 options here I guess:
1. We make the `controller.filter` public
2. We make a comment that if you are using a filter block, you need to use `replaceChannelListController()`

What do you think we should do? @ipavlidakis @polqf

Reply:
I think adding a comment will be ok for now.